src/braket/circuits/circuit.py

```diff
@@ -1089,13 +1096,52 @@ def as_unitary(self) -> np.ndarray:
                    [ 0.70710678+0.j, -0.70710678+0.j,  0.        +0.j,
                      0.        +0.j]])
         """
+        warnings.warn(
```

Comment:
Could we add `category=DeprecationWarning` to warnings.warn ?

Reply:
Good call!